lib/storage/metricnamestats/tracker.go

```diff
@@ -74,19 +74,20 @@ type recordForStore struct {
 // MustLoadFrom inits tracker from the given on-disk path
 func MustLoadFrom(loadPath string, maxSizeBytes uint64) *Tracker {
 	mt, err := loadFrom(loadPath, maxSizeBytes)
-	if err != nil {
-		logger.Fatalf("unexpected error at tracker state load from path=%q: %s", loadPath, err)
+	if err == nil {
+		return mt
 	}
-	return mt
+	if errors.Is(err, os.ErrNotExist) {
+		logger.Infof("tracker store file at path %s missing files; init new tracker", loadPath)
+	} else {
```

Comment:
This branch seems to be not reachable as `loadFrom` does not propagate `os.ErrNotExist` to the caller - https://github.com/VictoriaMetrics/VictoriaMetrics/blob/c3ba4a420793b55efc617b61be211d4cb9092e8d/lib/storage/metricnamestats/tracker.go#L92-L99

In case of `os.ErrNotExist` it `loadFrom` will return a new tracker.

Reply:
Thanks! good catch